lib/iris/fileformats/netcdf.py

```diff
@@ -1818,29 +2051,149 @@ def _get_coord_variable_name(self, cube, coord):
-                # Auto-generate name based on associated dimensions.
-                name = ""
-                for dim in cube.coord_dims(coord):
-                    name += "dim{}".format(dim)
-                # Handle scalar coordinate (dims == ()).
-                if not name:
-                    name = "unknown_scalar"
+                # We need to invent a name, based on its associated dimensions.
+                if cube.coords(coord):
+                    # It is a regular cube coordinate.
+                    # Auto-generate a name based on the dims.
+                    name = ""
+                    for dim in cube.coord_dims(coord):
+                        name += "dim{}".format(dim)
+                    # Handle scalar coordinate (dims == ()).
+                    if not name:
+                        name = "unknown_scalar"
+                else:
+                    # Not a cube coord, so must be a connectivity or
+                    # element-coordinate of the mesh.
+                    # Name it for it's first dim, i.e. mesh-dim of its location.
+                    from iris.experimental.ugrid import Connectivity
+
+                    if isinstance(coord, Connectivity):
+                        location = coord.cf_role.split("_")[0]
+                    else:
+                        # Must be a mesh-element coordinate.
+                        location = None
+                        for test_location in MESH_LOCATIONS:
+                            include_key = f"include_{test_location}s"
+                            if coord in cube.mesh.coords({include_key: True}):
+                                location = test_location
+                                break
+                        assert location is not None
+                    location_dim_attr = f"{location}_dimension"
+                    name = getattr(cube.mesh, location_dim_attr)
+
             # Convert to lower case and replace whitespace by underscores.
             cf_name = "_".join(name.lower().split())
 
         cf_name = self.cf_valid_var_name(cf_name)
         return cf_name
 
-    _ELEMENT_TYPE_NAMES = {
-        iris.coords.DimCoord: "coordinate",
-        iris.coords.AuxCoord: "coordinate",
-        iris.coords.CellMeasure: "cell-measure",
-        iris.coords.AncillaryVariable: "ancillary-variable",
-    }
+    def _get_mesh_variable_name(self, mesh):
+        """
+        Returns a CF-netCDF variable name for the mesh.
+
+        Args:
+
+        * mesh (:class:`iris.experimental.ugrid.Mesh`):
+            An instance of a Mesh for which a CF-netCDF variable name is
+            required.
+
+        Returns:
+            A CF-netCDF variable name as a string.
+
+        """
+        cf_name = mesh.var_name
+        if mesh.var_name is None:
+            # Prefer var-name, but accept long/standard as aliases.
+            cf_name = mesh.var_name or mesh.standard_name or mesh.long_name
```

Comment:
@pp-mo `mesh.var_name` is already `None` here... and personally I'd probably stray away from using the `standard_name` and `long_name`.

Why not just simply go for `Mesh_{...}d` here?

Reply:
> mesh.var_name is already None here

Whoops, yes !

> personally I'd probably stray away from using the standard_name and long_name... Why not just simply go for Mesh_{...}d 

I think it's entirely logical that a Mesh can never really have a standard_name, or units for that matter.
I will fix this.
The alternative mechanism seems sensible in itself, **_but_** I was trying to be consistent with usage elsewhere.  E.G. coords, cell measures etc will get a varname from the long-name if there is one.  So I think I would prefer to keep it doing that on "least surprise" grounds.